src/handlers/web3.ts

```diff
@@ -0,0 +1,689 @@
+import { getAddress } from '@ethersproject/address';
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { isHexString as isEthersHexString } from '@ethersproject/bytes';
+import { Contract } from '@ethersproject/contracts';
+import { isValidMnemonic as ethersIsValidMnemonic } from '@ethersproject/hdnode';
+import {
+  Block,
+  StaticJsonRpcProvider,
+  TransactionRequest,
+  TransactionResponse,
+} from '@ethersproject/providers';
+import { parseEther } from '@ethersproject/units';
+import UnstoppableResolution from '@unstoppabledomains/resolution';
+import { startsWith } from 'lodash';
+import { RainbowConfig } from '../model/config';
+import {
+  AssetType,
+  NewTransaction,
+  ParsedAddressAsset,
+} from '@rainbow-me/entities';
+import { isNativeAsset } from '@rainbow-me/handlers/assets';
+import { Network } from '@rainbow-me/helpers/networkTypes';
+import {
+  ARBITRUM_ETH_ADDRESS,
+  ETH_ADDRESS,
+  ethUnits,
+  MATIC_POLYGON_ADDRESS,
+  OPTIMISM_ETH_ADDRESS,
+  smartContractMethods,
+} from '@rainbow-me/references';
+import {
+  addBuffer,
+  convertAmountToRawAmount,
+  convertStringToHex,
+  fraction,
+  greaterThan,
+  handleSignificantDecimals,
+  multiply,
+} from '@rainbow-me/utilities';
+import { ethereumUtils } from '@rainbow-me/utils';
+import logger from 'logger';
+
+export const networkProviders: {
+  [network in Network]?: StaticJsonRpcProvider;
+} = {};
+
+const rpcEndpoints: { [network in Network]?: string } = {};
+
+/**
+ * The format of transaction details returned by functions such as `getTxDetails`.
+ */
+interface TransactionDetailsReturned {
+  data?: TransactionRequest['data'];
```

Comment:
can we alphabetize the params in this interface

Reply:
Sounds great, I pushed a change alphabetizing them. Thanks!